### PR TITLE
add seeking to passthrough

### PR DESCRIPTION
**Pull request by philippe44**
Let me know if this works for you. It seems to work on my x86 machine. 

**important**

The --pass-through option is now --passthrough (I know ...)

Don't forget to change the custom-convert.conf as well : smile:

### CODE REVIEW

**Comment by michaelherger** [comment]
QQ: what's the minimum Rust version required for this branch?

**Comment by michaelherger** [comment]
I'm on rust 1.51.0 and am getting a lot of these warnings:

```
warning: trait objects without an explicit `dyn` are deprecated
  --> core/src/keymaster.rs:20:10
   |
20 | ) -> Box<Future<Item = Token, Error = MercuryError>> {
   |          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: use `dyn`: `dyn Future<Item = Token, Error = MercuryError>`
```

Do you see this, too? Or am I ahead of what you've been using?

**Comment by philippe44** [comment]
Yes, there is a fair bit of warning, but I'm not sure why. I thought they were there before. I've not changed many files, as you can see, maybe just the versions of dependencies? That might be the culprit. It caused me a fair bit of pain to figure out that one compiler error was with protobuf. Although it is, in librespot's cargo.toml, set to 2.8.1, it means that any >2.8.1 is supposed to work but it's not the the case. I had to learn about cargo dependencies and force it to be either = or ~ to 2.8.1

File: audio/Cargo.toml [hunk 1]
```diff
@@ -4,24 +4,28 @@ version = "0.1.0"
...
+num-bigint = "0.3"
 num-traits = "0.2"
 tempfile = "3.1"
-aes-ctr = "0.3"
+backtrace = "0.3"
 
-librespot-tremor = { version = "0.1.0", optional = true }
-vorbis = { version ="0.1.0", optional = true }
+librespot-tremor = { version = "0.2.0", optional = true }
+vorbis = { version ="0.0.14", optional = true }
```
Comment (michaelherger) [inline audio/Cargo.toml]: This seems like a step backwards? Why is that?

File: playback/src/player.rs [hunk 2]
```diff
@@ -387,20 +388,20 @@ impl PlayerInternal {
     fn handle_packet(&mut self, packet: Option<AudioPacket>, normalisation_factor: f32) {
         match packet {
             Some(mut packet) => {
-                if packet.data().len() > 0 {
-                    if !self.config.pass_through {
+				if !packet.is_empty() {
+					if let AudioPacket::Samples(ref mut data) = packet {
```
Comment (michaelherger) [inline playback/src/player.rs]: whitespace

File: playback/src/player.rs [hunk 3]
```diff
@@ -387,20 +388,20 @@ impl PlayerInternal {
...
                         if self.config.normalisation && normalisation_factor != 1.0 {
-                            for x in packet.data_mut().iter_mut() {
+                            for x in data.iter_mut() {
                                 *x = (*x as f32 * normalisation_factor) as i16;
                             }
                         }
                     }
 
-                    if let Err(err) = self.sink.write(&packet.data()) {
+					if let Err(err) = self.sink.write(&packet) {
```
Comment (michaelherger) [inline playback/src/player.rs]: whitespace

File: audio/src/lewton_decoder.rs [hunk 4]
```diff
@@ -17,29 +17,6 @@ where
...
     }
-
-    pub fn seek(&mut self, ms: i64) -> Result<(), VorbisError> {
-        let absgp = ms * 44100 / 1000;
-        self.0.seek_absgp_pg(absgp as u64)?;
-        Ok(())
-    }
-
-    pub fn next_packet(&mut self) -> Result<Option<AudioPacket>, VorbisError> {
-        use self::lewton::audio::AudioReadError::AudioIsHeader;
```
Comment (michaelherger) [inline audio/src/lewton_decoder.rs:28]: Why can you remove all this code? The function name suggests it's being used for seeking :-)

Is lewton even the decoder we're using?...

**Review by michaelherger — COMMENTED** [review]
I'll have to trust your skills here :-).

I was a little surprised by quite a bit of code you removed from the decoders which seems to be relevant to seeking? Is this no longer being used?

File: audio/Cargo.toml (hunk 1 above)
Comment (philippe44) [reply audio/Cargo.toml]: It's the master of librespot that has even moved to 0.0 which I think is 'don't care"

File: playback/src/player.rs (hunk 2 above)
Comment (philippe44) [reply playback/src/player.rs]: done

File: audio/src/lewton_decoder.rs (hunk 4 above)
Comment (philippe44) [reply audio/src/lewton_decoder.rs:28]: It was a duplication left at the origin. You can see that seek and next_packet exist twice. By default, lewton is the decoder used unless you set differently your build command line (I don't know what it is you use)

File: audio/Cargo.toml (hunk 1 above)
Comment (philippe44) [reply audio/Cargo.toml]: I should add the lewton is the one used by default, not vorbis, so unless spotty has a different command line, it should not even matter

**Comment by michaelherger** [comment]
🙇🏻

**Pull request merged**